### PR TITLE
Add Custom control server & fix serve command

### PR DESCRIPTION
**Pull request by BehnH**
This PR adds support for the environment variable `TAILSCALE_CUSTOM_SERVER` so custom control planes (Such as Headscale) can be used.

Also fixes the `tailscale serve` command always serving port 443, and not actually working if `http` is set as the TAILSCALE_SERVE_MODE` value

### CODE REVIEW

**Comment by B08Z** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ use to configure it.
...
 | `TAILSCALE_SERVE_MODE` | The mode you want to run Tailscale serving in. This should be `https` in most cases, but there may be times when you need to enable `tls-terminated-tcp` to deal with some weird edge cases like HTTP long-poll connections. See [here](https://tailscale.com/kb/1242/tailscale-serve/) for more information. | `https`                                  |
 | `TAILSCALE_FUNNEL`     | Set this to `true`, `1`, or `t` to enable [funnel](https://tailscale.com/kb/1243/funnel/). For more information about the accepted syntax, please read the [strconv.ParseBool documentation](https://pkg.go.dev/strconv#ParseBool) in the Go standard library.                                                | `on`                                     |
+| `TAILSCALE_CUSTOM_SERVER` | Set this value if you are using a custom login/control server (Such as headscale) | `https://headscale.example.com`
```
Comment (tylersmalley) [inline README.md:22]: Can we change the env to `TAILSCALE_LOGIN_SERVER` to reflect the setting?

**Comment by BehnH** [comment]
Sorry this slipped my radar - I'll jump on it this weekend!

File: README.md (hunk 1 above)
Comment (BehnH) [reply README.md]: Renamed env var as requested

**Comment by alexfornuto** [comment]
> Sorry this slipped my radar - I'll jump on it this weekend! Looks like there are some conflicts I need to fix up too

Thanks, but I think at this point the project is abandonware waiting to be called as such.

**Comment by c0re16** [comment]
+1, I would use the heck out of this docker mod if headscale was supported!

**Review by tylersmalley — APPROVED** [review]

**Pull request merged**